### PR TITLE
Make adding the hostname to the event optional

### PR DESCRIPTION
**Pull request by robbavey**
Add option to make the population of the 'host' field in the event with
the value from `beat.hostname` optional. This field is intentionally added
as a deprecated field, as the this field will be removed in future versions
of the beats input, when the ability to populate `host` field in this way
will be removed altogether.

### CODE REVIEW

File: lib/logstash/inputs/beats.rb [hunk 1]
```diff
@@ -82,6 +82,10 @@ class LogStash::Inputs::Beats < LogStash::Inputs::Base
   # 
   config :ssl_certificate_authorities, :validate => :array, :default => []
 
+  # Flag to determine whether to add host information (provided by the beat in the 'hostname' field) to the event
+  config :add_hostname, :validate => :boolean, :default => true, :deprecated => 'Host field will not be automatically populated by future version of the Beats input'
```
Comment (danhermann) [inline lib/logstash/inputs/beats.rb]: Doesn't the default value of `true` conflict with the documented default value?

Comment (robbavey) [reply lib/logstash/inputs/beats.rb]: Gah, you're right. Good catch, will change

**Comment by robbavey** [comment]
@karenzone Would you mind checking the doc changes?

File: docs/index.asciidoc [hunk 2]
```diff
@@ -112,6 +113,18 @@ input plugins.
 
 &nbsp;
 
+[id="plugins-{type}s-{plugin}-add_hostname"]
+===== `add_hostname`
+
+  added[5.2.0, Field was added to allow users to not automatically add the `host` field in events.]
```
Comment (karenzone) [inline docs/index.asciidoc]: What about "Field was added to allow users to control whether or not the `host` field is automatically added to events."

File: docs/index.asciidoc [hunk 3]
```diff
@@ -112,6 +113,18 @@ input plugins.
...
+  deprecated[5.2.0, In future versions of this plugin, this setting will be removed, and the 'hosts' field will not be added to events.]
```
Comment (karenzone) [inline docs/index.asciidoc]: Can we say `deprecated[5.2.1` if the field will be removed for *future* releases? 
To summarize suggestion, "Added in 5.2.0.  Deprecated in 5.2.1"

File: docs/index.asciidoc (hunk 2 above)
Comment (robbavey) [reply docs/index.asciidoc]: +1 - I like that

File: docs/index.asciidoc (hunk 3 above)
Comment (robbavey) [reply docs/index.asciidoc]: The intent is to deprecate it immediately - the setting is a stop gap until we make the breaking change that will remove the setting.

**Pull request closed without merging**